src/sidebar/services/frame-sync.js

```diff
@@ -50,63 +50,67 @@ export default function FrameSync($rootScope, $window, store, bridge) {
    * notify connected frames about new/updated/deleted annotations.
    */
   function setupSyncToFrame() {
-    let prevPublicAnns = 0;
+    let prevPublicAnnotationCount = 0;
 
     watch(
       store.subscribe,
-      [() => store.getState().annotations.annotations, () => store.frames()],
+      [
+        () => store.getState().annotations.annotations,
+        () => store.frames(),
+        () => store.frameFetchStatus(),
+      ],
       ([annotations, frames], [prevAnnotations]) => {
-        let publicAnns = 0;
-        const inSidebar = new Set();
-        const added = [];
+        let publicAnnotationCount = 0;
```

Comment:
Changes here are primarily lightweight refactor for readability and modern syntax.